ecosystem/typescript/sdk/src/plugins/coin_client.ts

```diff
@@ -54,8 +60,27 @@ export class CoinClient {
       // If this is the first time an account has received the specified coinType,
       // and this is set to false, the transaction would fail.
       createReceiverIfMissing?: boolean;
+      // fungible asset address if you wish to transfer a fungible asset
+      fungibleAssetMetadataAddress?: MaybeHexString;
```

Comment:
I don't really see the benefit of having this here where people need to call coin_client.transfer with different params anyway vs just calling fungible_asset_client directly. I was thinking that if we can combine this with coinType, at least the benefit would be that people can still use the same function coin_client.transfer with same exact inputs and args, except that coinType would be an address (instead of a struct type) as these data would be coming from indexer queries. Can you think through this flow end to end?